File: C++/49-把字符串转换为整数.cpp
class Solution
{
public:
    int StrToInt(string str)
    {
        bool flag = true;
        int res = 0, base = 1, start = 0, idx = str.size() - 1;
        if (str[0] == '+' || str[0] == '-')
        {
            start++;
            if (str[0] == '-')
                flag = false;
        }
        while (idx >= start)
        {
            if (str[idx] >= '0' && str[idx] <= '9')
            {
                res += base * (str[idx] - '0');
                base *= 10;
                idx--;
            }
            else
                return 0;
        }
        return flag ? res : -res;
    }
};

File: Python/49-把字符串转换为整数.py
# -*- coding:utf-8 -*-
class Solution:
    def StrToInt(self, s):
        # write code here
        flag = True
        start = 0
        res = 0
        if not s:
            return 0
        if s[0] in '+-':
            start = 1
            if s[0] == '-':
                flag = False
        base = 1
        idx = len(s) - 1
        while idx >= start:
            if s[idx].isdigit():
                res += base * int(s[idx])
                base *= 10
                idx -= 1
            else:
                return 0
        return res if flag else -res